cmd/resourcegen/main.go

```diff
@@ -0,0 +1,304 @@
+package main
+
+import (
+	"embed"
+	"errors"
+	"flag"
+	"fmt"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"os"
+	"path"
+	"regexp"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/dave/dst"
+	"github.com/dave/dst/decorator"
+)
+
+var (
+	//go:embed embed/*
+	assets embed.FS
+
+	allowedProviders = map[string]struct{}{
+		"aws":    {},
+		"azure":  {},
+		"google": {},
+	}
+
+	embedPathExp  = regexp.MustCompile(`^embed/`)
+	tmplSuffixExp = regexp.MustCompile(`\.tmpl$`)
+
+	renderDir = "internal"
+)
+
+func main() {
+	var c config
+	flag.StringVar(&c.CloudProvider, "cloud-provider", "aws", "Cloud provider to create resource for, one of [aws, azure, google]")
+	flag.StringVar(&c.Filename, "resource-name", "", "The resource name to generate, use underscores between names, e.g. autoscaling_group (required)")
+	flag.BoolVar(&c.WithHelp, "with-help", true, "Generate your resources with doc blocks and examples to help you get started. Useful for understanding how to add a resource")
+	flag.Parse()
+
+	if flag.NFlag() == 0 {
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
```

Comment:
@hugorut Tried to address Ali's comment (I guess we can live with single dash flags for now without introducing cobra), but ended up adding some extra formatting changes (see last commit). Hope it helps, feel free to change or throw away. :)

Reply:
the go default flags package allows both `--` and `-` to define the flags so double dash flags should work out the box. See the example image in the PR description (has `--` flags). Changes lGTM 👍

Reply:
Yeah, sorry, I meant we don't need to enforce the double dashes. Thanks for review! :)